src/dialog/provider.js

```diff
@@ -18,107 +18,72 @@ export default class FlowProvider extends EventEmitter2 {
+        }
 
-      if (fs.existsSync(uiEqPath)) {
-        uiEq = JSON.parse(fs.readFileSync(uiEqPath))
-      }
+        const uiEq = JSON.parse(await this.ghostManager.readFile(this.flowsDir, uiFileName))
 
-      Object.assign(flow, { links: uiEq.links })
+        Object.assign(flow, { links: uiEq.links })
 
-      // Take position from UI files or create default position
-      const unplacedNodes = []
-      flow.nodes.forEach(node => {
-        const uiNode = _.find(uiEq.nodes, { id: node.id }) || {}
+        // Take position from UI files or create default position
+        flow.nodes.forEach(node => {
+          const uiNode = _.find(uiEq.nodes, { id: node.id }) || {}
+          Object.assign(node, uiNode.position)
+        })
 
-        Object.assign(node, uiNode.position)
+        const unplacedY = (_.maxBy(flow.nodes, 'y') || { y: 0 }).y + 250
+        flow.nodes.filter(node => _.isNil(node.x) || _.isNil(node.y)).forEach((node, i) => {
+          node.y = unplacedY
+          node.x = 50 + i * 250
+        })
 
-        if (_.isNil(node.x) || _.isNil(node.y)) {
-          unplacedNodes.push(node)
+        return {
+          name,
+          location: name,
+          nodes: _.filter(flow.nodes, node => !!node),
+          ..._.pick(flow, 'version', 'catchAll', 'startNode', 'links', 'skillData')
         }
       })
+    )
 
-      const unplacedY = (_.maxBy(flow.nodes, 'y') || { y: 0 }).y + 250
-      let unplacedX = 50
-
-      unplacedNodes.forEach(node => {
-        node.y = unplacedY
-        node.x = unplacedX
-        unplacedX += 250
-      })
-
-      return flows.push({
-        location: file,
-        version: flow.version,
-        name: flow.name,
-        nodes: _.filter(flow.nodes, node => !!node),
-        catchAll: flow.catchAll,
-        startNode: flow.startNode,
-        links: flow.links,
-        skillData: flow.skillData
-      })
-    })
-
-    return flows
+    return flows.filter(flow => Boolean(flow))
   }
 
   async saveFlows(flows) {
     const flowsToSave = await Promise.mapSeries(flows, flow => this._prepareSaveFlow(flow))
 
     for (const { flowPath, uiPath, flowContent, uiContent } of flowsToSave) {
-      if (flowPath.includes('/')) {
-        mkdirp.sync(path.dirname(flowPath))
-      }
-
-      fs.writeFileSync(flowPath, JSON.stringify(flowContent, null, 2))
-      fs.writeFileSync(uiPath, JSON.stringify(uiContent, null, 2))
+      this.ghostManager.upsertFile(this.flowsDir, flowPath, JSON.stringify(flowContent, null, 2))
+      this.ghostManager.upsertFile(this.flowsDir, uiPath, JSON.stringify(uiContent, null, 2))
     }
 
-    const flowsDir = path.resolve(this.projectLocation, this.botfile.flowsDir || './flows')
-
-    const searchOptions = { cwd: flowsDir }
-    const flowFiles = await Promise.fromCallback(callback => glob('**/*.flow.json', searchOptions, callback))
-
+    const flowFiles = await this.ghostManager.directoryListing(this.flowsDir, '.json')
     flowFiles
```

Comment:
looks like we're not awaiting for this operation to end?

Reply:
Yep, fixed that.